app/ts/background/windows/interceptorAccess.ts

```diff
@@ -225,6 +231,10 @@ async function resolve(simulator: Simulator, websiteTabConnections: WebsiteTabCo
 		await closePopupOrTab(openedDialog.popupOrTab)
 		openedDialog = undefined
 	}
+
+	await Promise.all(affected.map(async (r) => {
+		if (r.request) await handleInterceptedRequest(undefined, r.website.websiteOrigin, r.website, simulator, r.socket, r.request, websiteTabConnections)
```

Comment:
It feels like this conditional should be extracted out into a `.filter`  That would allow you to remove the `async` and `await` and curly braces as you can just turn this lambda into an expression.

Reply:
I don't think that's cleaner as filter doesn't manage the typing right and then we need to use `is` and construct a type that has that variable as non-undefined